Added ability to open/import .csv files with # comments in them (no use of temp files).

NGD-Skier opened the pull request:
This is a new pull request to replace PR 870, because I couldn't figure out how to get rid of the "Please do not include merge commits in your PR" (tried rebasing, etc)

Here's there original discussion regarding PR 870, for reference:
[Please do not include merge commits in your PR](https://github.com/kk7ds/chirp/pull/870)

This is related to feature request 11081:
https://chirp.danplanet.com/issues/11081

Anyway...

* Rewrote to NOT use a temp file to strip out # comment lines from opened/imported .csv files.

Now, comments are stripped in two places:

-chirp/directory.py:  where lines starting with "#" are stripped out from filedata variable after a binary read.  If the file read in is a .csv file, then filedata is split into lines and then rejoined, skipping lines that start with "#", if it's not a .csv file, but nothing is modified from the usual code path.

-chirp/drivers/generic_csv.py:  Where lines are read in to "line" by csv.reader and lines who's first element starts with "#" are skipped (i.e. jump to next iteration of the loops without being processed)

You can open and import .csv files that have comment lines that start with "#" anywhere in the file.

kk7ds reviewed (changes requested):
Please don't open a new PR like this, as it's nice to keep all the discussion about a change together for posterity. You need to rebase and squash your commits into one for each logical change anyway (force push to update the PR). If you can't figure it out just push it up with the failing check and I'll fix it for you before merge.

You also need:
- The line "Fixes #11081" in your commit message so the bug system will track and link it when it merges
- The first line of your commit message needs to explain the change in just a few characters because that gets sent to the users list on every build. So just "Allow comment lines in CSV files" would be perfect.

No need to update the chirp issue tracker with each rev of this PR, it'll get auto-updated when it's merged if you have the proper fixes notation in the commit message.

Thanks!

kk7ds commented:
I never heard back, so I applied the above change and cleaned this up. Hope that's okay. I also added persistence so it *should* prevent losing comments if you open/save in chirp.

NGD-Skier commented:
> I never heard back, so I applied the above change and cleaned this up. Hope that's okay. I also added persistence so it _should_ prevent losing comments if you open/save in chirp.

No worries.  I just got too busy with work and family to spend any time on it.  Nice work on adding persistence.  I did a little testing with it and all seemed to be working.  Thanks!  Sorry my initial efforts weren't quite in line with all the requirement to make the pull request acceptable, but it's great that the feature got in there.  Will it be a while before this update is released for general use, or will the latest build have this in there now?

kk7ds commented:
Yep, it's released. Anything I merge gets released the following day automatically.

And no worries on the "acceptable" stuff, I'm glad to have the feature in there, so thanks for suggesting it :)
